src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Routes/RouteService.cs

```
@@ -24,6 +24,25 @@ protected override System.Data.Entity.IDbSet<Route> Table
             get { return Context.Routes; }
         }
 
+        public override Route Create(Route entity)
+        {
+            if (Table.AsNoTracking().SingleOrDefault(r => r.Slug == entity.Slug) != null)
+            {
+                throw new Exception("Slug '" + entity.Slug + "' is existing");
+            }
+            return base.Create(entity);
+        }
+
+        public override Route Update(Route entity)
+        {
+            if (Table.AsNoTracking().SingleOrDefault(r => r.Slug == entity.Slug && r.Id != entity.Id) != null)
```

This will throw an exception if such route already exists, but it should throw an exception if there is no route with such `Id`